src/Control/Monad/Logic/Sequence/Internal.hs

```diff
@@ -325,6 +327,17 @@ instance Monad m => MonadLogic (SeqT m) where
       Empty -> single Nothing
       a :< t -> single (Just (a, t))
 
+-- | @choose = foldr (\a s -> pure a <|> s) empty@
+choose :: Monad m => [a] -> SeqT m a
+choose = foldr (\a s -> fromView (pure (a :< s))) empty
```

Comment:
I think it would be better to use `F.foldr` here so this works for any `Foldable` instead of `[]`.

Reply:
You probably already figured this out, but due to supporting pre-AMP GHC's we have to use `return` instead of `pure`.